draft-ietf-cdni-ci-triggers-rfc8007bis.xml

```diff
@@ -3456,808 +3548,1280 @@ ObjectLists MAY combine regular objects and secondary ObjectLists in the same ob
+Host: dcdn.example
+Accept: */*
+Content-Type: application/cdni; ptype=ci-trigger.v2
+Content-Length: 26
+
+{
+  "state":"cancelled"
+}
+
+RESPONSE: 
+
+HTTP/1.1 20O OK
+Date: Sun, 27 Oct 2024 08:48:15 GMT
+Content-Length: 523
+Content-Type: application/cdni; ptype=ci-trigger.v2
+Server: example-server/0.1
+
+{
+  "ctime": 1730119695,
+  "etime": 1730119755,
+  "mtime": 1730119695,
+  "state": "cancelling",
+  "action": "preposition",
+
+  "specs": [
+    {
+      "trigger-subject": "content",
+      "generic-trigger-spec-type": "urls",
+      "generic-trigger-spec-value": {
+        "urls": [
+          "https://www.example.com/d/e/f/1",
+          "https://www.example.com/d/e/f/2",
+          "https://www.example.com/d/e/f/3",
+          "https://www.example.com/d/e/f/4"
+        ]
+      }
+    }
+  ],
+  "labels": [
+    "type=video"
+  ]
+}
+]]></artwork></figure>
+    </section>
+
+    <section title="Deleting Triggers" anchor="examples-deleting-trigger">
+    <t>
+    uCDN can delete completed and failed triggers to reduce the size of the collections, as described in 
+    <xref target="deleting-triggers"/>.  For example, to delete the "preposition" trigger from earlier examples:
+    </t>
 
    <figure><artwork><![CDATA[
 REQUEST:
 
-  GET /triggers HTTP/1.1
-  User-Agent: example-user-agent/0.1
-  Host: dcdn.example.com
-  Accept: */*
+DELETE /cit/3f2d259d-a980-4742-beeb-9392a58129f5 HTTP/1.1
+User-Agent: example-user-agent/0.1
+Host: dcdn.example
+Accept: */*
+
+RESPONSE:
+
+HTTP/1.1 204 No Content
+Date: Sun, 27 Oct 2024 08:48:16 GMT
+Content-Length: 0
+Content-Type: text/html; charset=UTF-8
+Server: example-server/0.1
+]]></artwork>
+   </figure>
+   <t>
+   This would, for example, cause the collection of completed Trigger
+   Status Resources shown in the example above, to be updated to:</t>
+
+   <figure><artwork><![CDATA[
+REQUEST:
+
+GET /cit/state/complete HTTP/1.1
+User-Agent: example-user-agent/0.1
+Host: dcdn.example
+Accept: */*
+
+RESPONSE:
+
+HTTP/1.1 200 OK
+Content-Length: 122
+Expires: Sun, 27 Oct 2024 08:49:17 GMT
+Server: example-server/0.1
+ETag: "6938620031669085677"
+Last-Modified: Sun, 27 Oct 2024 08:48:17 GMT
+Cache-Control: max-age=60
+Date: Sun, 27 Oct 2024 08:48:17 GMT
+Content-Type: application/cdni; ptype=ci-trigger-collection.v2
+
+{
+  "staleresourcetime": 86400,
+  "triggers": [ 
+    "https://dcdn.example/cit/327df5b8-1df8-4cff-92f8-fda27774c171"
+  ]
+}
+]]></artwork>
+    </figure>
+    </section>
+  </section>
+  <section title="Examining Trigger Status" anchor="examples-examining-trigger-status">
+  <t>
+  Once triggers have been created, uCDN can check their status as shown in the following examples.
+  </t>
+
+    <section title="Collection of All Triggers" anchor="examples-collections-of-all-triggers">
+    <t>
+    uCDN can fetch the collection of all triggers it has created that have not yet been deleted or expired.
+    After the creation of the "preposition" and "invalidate" triggers shown
```

Comment:
This statement is kind of inaccurate now, since it is technically the "Preposition with Objectlists" and "Invalidate with Regex" examples only, but not the more ordinary "Preposition" and "Invalidate" triggers.  Should we add their URIs to the list as well for completion?

Reply:
rewrote to say:
   After the trigger operations described in <xref target="examples-creating-triggers"/> and
    <xref target="examples-changing-triggers"/> above, this collection might look as follows: